### PR TITLE
Feat: Update Android + clean up Android

### LINKED ISSUE #470: Use latest options from hub instead of using private options set on init
This would not be a major issue for Flutter now as it does not look like there is an equivalent of `autoInitNativeSdk` option here, but there was an issue over on RN regarding the user initing the native SDK themselves elsewhere and the outbox path not being correctly retrieved when writing envelopes causing error handling to not work.

Relevant issues and PR over on RN: https://github.com/getsentry/sentry-react-native/pull/1516 https://github.com/getsentry/sentry-react-native/pull/1529

https://github.com/getsentry/sentry-dart/blob/bece2c266231f291e970c667fb203a42951de538/flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt#L49-L58

### LINKED ISSUE #481: Remove thread clean up for Android
Related to https://github.com/getsentry/sentry-java/issues/1473

Need to publish and bump the Android SDK

### PR DESCRIPTION
**Pull request by ueman**
## :scroll: Description
- Update Sentry Android to 5 GA
- Remove Thread related code
- Update Android from cacheDirSize to maxCacheItems (also created https://github.com/getsentry/sentry-dart/issues/485 for Dart)

## :bulb: Motivation and Context
Closes https://github.com/getsentry/sentry-dart/issues/470
Closes https://github.com/getsentry/sentry-dart/issues/481

## :green_heart: How did you test it?
https://sentry.io/organizations/sentry-sdks/issues/2149575550/?project=5428562&query=is%3Aunresolved

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 1]
```diff
@@ -110,13 +111,11 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
       options.setBeforeSend { event, _ ->
         setEventOriginTag(event)
         addPackages(event, options.sdkVersion)
-        removeThreadsIfNotAndroid(event)
         // TODO: merge debug images from Native
```
Comment (ueman) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: Do we already have an issue for this?

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 2]
```diff
@@ -110,13 +111,11 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
...
         event
       }
 
-      // missing proxy, sendDefaultPii, enableScopeSync
-      this.options = options
```
Comment (ueman) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt:119]: `HubAdapter.getInstance().options` replaces this

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 3]
```diff
@@ -87,7 +88,7 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
         options.isEnableAppComponentBreadcrumbs = it
       }
       args.getIfNotNull<Int>("maxBreadcrumbs") { options.maxBreadcrumbs = it }
-      args.getIfNotNull<Int>("cacheDirSize") { options.cacheDirSize = it }
+      args.getIfNotNull<Int>("cacheDirSize") { options.maxCacheItems = it }
```
Comment (marandaneto) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt:91]:
```suggestion
      args.getIfNotNull<Int>("maxCacheItems") { options.maxCacheItems = it }
```

we should fix the param name passing as well, also on iOS which is already called maxCacheItems I guess

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # Unreleased
 
 * Fix: Merge user from event and scope (#467)
+* Feat: Clean up Android implementation (#486)
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
```

I'd just keep the bump as the changes are because of it anyway

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Unreleased
 
 * Fix: Merge user from event and scope (#467)
+* Bump: sentry-android to v5.0.0 (#486)
```
Comment (marandaneto) [inline CHANGELOG.md]:
```suggestion
* Bump: sentry-android to v5.0.1 (#486)
```

File: flutter/android/build.gradle [hunk 6]
```diff
@@ -56,6 +56,6 @@ android {
 }
 
 dependencies {
-    api 'io.sentry:sentry-android:5.0.0-beta.2'
+    api 'io.sentry:sentry-android:5.0.0'
```
Comment (marandaneto) [inline flutter/android/build.gradle]:
```suggestion
    api 'io.sentry:sentry-android:5.0.1'
```

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt (hunk 3 above)
Comment (marandaneto) [reply flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt:91]: sounds good

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt (hunk 1 above)
Comment (marandaneto) [reply flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: no, and I don't think it's relevant anymore, could be removed, iirc this is done by `loadImageList` method

**Comment by marandaneto** [comment]
bumped to 5.0.1 because of a bug we had on 5.0.0 https://github.com/getsentry/sentry-java/releases/tag/5.0.1

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt (hunk 1 above)
Comment (ueman) [reply flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]:
```suggestion
```

**Review by marandaneto — APPROVED** [review]
LGTM

**Pull request merged**